### PR TITLE
Fix json equality check

### PR DESCRIPTION
**Pull request by kriks57**
Previously, when using JSON Type for a column, every setXXX() on this column would result in a modified column.
This PR adds a check between old JSON and new value, as objects.

### CODE REVIEW

File: src/Propel/Generator/Builder/Om/ObjectBuilder.php [hunk 1]
```diff
@@ -999,11 +1001,32 @@ public function addJsonAccessorComment(&$script, Column $column)
...
+     * @param string &$script
+     * @param Column $column
+     */
+    public function addJsonAccessorOpen(&$script, Column $column)
+    {
+        $cfc = $column->getPhpName();
+        $visibility = $column->getAccessorVisibility();
+
+        $script .= "
+    ".$visibility." function get$cfc(\$asArray = false";
```
Comment (marcj) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: I'd go per default asArray=true

Comment (kriks57) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: I would too, but it would change previous behavior

Comment (marcj) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: do it :)

Comment (kriks57) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: BTW, it's done :)

File: src/Propel/Generator/Builder/Om/ObjectBuilder.php [hunk 2]
```diff
@@ -1804,11 +1881,19 @@ protected function addJsonMutator(&$script, Column $col)
...
         $clo = $col->getLowercasedName();
 
-        $this->addMutatorOpen($script, $col);
+        $this->addJsonMutatorOpen($script, $col);
 
         $script .= "
-        \$this->$clo = json_encode(\$v);
-        \$this->modifiedColumns[".$this->getColumnConstant($col)."] = true;
+        if (is_array(\$v)) {
+            \$v = json_decode(json_encode(\$v));
```
Comment (marcj) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: what purpose has this line?

File: composer.json [hunk 3]
```diff
@@ -43,7 +43,8 @@
     ],
     "extra": {
         "branch-alias": {
-            "dev-master": "2.0-dev"
+            "dev-master": "2.0-dev",
+            "dev-fix-json-equality": "2.0-dev"
```
Comment (marcj) [inline composer.json]: we can remove that again

File: src/Propel/Generator/Builder/Om/ObjectBuilder.php (hunk 2 above)
Comment (kriks57) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: transform array to object, for camparison

File: composer.json (hunk 3 above)
Comment (kriks57) [inline composer.json]: just before merging, we currently need it :)

File: src/Propel/Generator/Builder/Om/ObjectBuilder.php (hunk 2 above)
Comment (staabm) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: couldnt you just cast it to `(object)`?

Comment (kriks57) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: it wouldn't work recursively

Comment (marcj) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: I don't understand. Why don't compare the actual object, but the json-string of it. Where is a object comparison happen?

Comment (kriks57) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: 5 lines below :)
`if (\$v != json_decode(\$this->$clo)) {`

Comment (marcj) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: Then we should change that to `if (json_encode($v) != $this->$clo)`, it's safer and probably faster.

Comment (kriks57) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: Let's try

Comment (kriks57) [inline src/Propel/Generator/Builder/Om/ObjectBuilder.php]: done, what do you think ?

**Comment by kriks57** [comment]
Could we merge ?

File: composer.json (hunk 3 above)
Comment (marcj) [inline composer.json]: Then remove it ;P

Comment (kriks57) [inline composer.json]: done :)

**Comment by kriks57** [comment]
Now, can we? :)

**Comment by marcj** [comment]
Alrighty, thanks!

**Pull request merged**